### ISSUE TITLE
feat: BatchNorm layer for tabular models

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#99]
## Summary

Batch normalization between dense layers significantly improves training stability and convergence speed for MLPs on tabular data, especially when input features have different scales (e.g., price returns at 0.001 scale vs volume at 1M scale).

## Proposed API

```go
// In package layers/normalization
type BatchNorm[T numeric.Float] struct {
    Features int
    Momentum float64 // running stats momentum (default 0.1)
    Eps      float64 // numerical stability (default 1e-5)
}

func NewBatchNorm[T numeric.Float](engine compute.Engine[T], ops numeric.Arithmetic[T], features int) *BatchNorm[T]
```

## Use Case

Wolf's COIN feature set includes values spanning 6+ orders of magnitude. Without normalization, gradients are dominated by large-scale features. BatchNorm after each dense layer would:
1. Stabilize gradients across feature scales
2. Enable higher learning rates
3. Act as mild regularization (batch noise)

## Typical MLP Architecture with BatchNorm

```
Input → Dense → BatchNorm → ReLU → Dropout → Dense → BatchNorm → ReLU → Dropout → Dense → Sigmoid
```
